core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/ReadOnlyLDAPUserStoreManager.java

```diff
@@ -482,23 +482,26 @@ public boolean doAuthenticate(String userName, Object credential) throws UserSto
                 }
             } else {
                 name = getNameInSpaceForUserName(userName);
-                try {
-                    if (name != null) {
-                        if (debug) {
-                            log.debug("Authenticating with " + name);
+                // if it is the same user DN found in the cache no need of futher authentication required.
+                if (failedUserDN != null && !failedUserDN.equalsIgnoreCase(name)) {
```

Comment:
There is another problem with this condition now.
failedUserDN reference will be null, if user dn is not used from cache and no user dn patterns configured.
In such a case the binding should happen for user dn searched from search base.
If it is not null and if the user dn searched from search base is same only, the binding should not happen.

Could you please fix the condition and the logic accordingly.